src/main/scala/org/broadinstitute/dsde/firecloud/service/FireCloudDirectives.scala

```diff
@@ -35,4 +37,14 @@ trait FireCloudDirectives extends spray.routing.Directives with PerRequestCreato
       }
     }
   }
+
+  def encodeUri(path: String): String = {
+    val pattern = """(https|http)://([^/\r\n]+)(/[^\r\n]*)?""".r
```

Comment:
That regex is just to split the string into the scheme, host, and path so that a uri can be formed from it. The forming of the uri is where the encoding takes place, which is done so automatically. I can share a relevant workspace with you so that you can check the before/after and make sure that everything works (it's called "mbemis space").